src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -107,8 +104,8 @@ protected function processValue($value, $isRoot = false)
             $autowiredMethods = $this->getMethodsToAutowire($reflectionClass);
             $methodCalls = $value->getMethodCalls();
 
-            if ($constructor = $reflectionClass->getConstructor()) {
-                array_unshift($methodCalls, array($constructor->name, $value->getArguments()));
+            if ($constructor = $this->getConstructor($value, false)) {
+                array_unshift($methodCalls, array($constructor, $value->getArguments()));
```

Comment:
nope we can't, because when a factory is used, a simple string is not enough to reference the target